src/components/C2Table.tsx

```diff
@@ -0,0 +1,143 @@
+import { ChevronDownIcon, ChevronUpIcon } from "@chakra-ui/icons";
+import {
+    FormControl,
+    Table,
+    Thead,
+    Tbody,
+    Tr,
+    Th,
+    Td,
+    Flex,
+    Icon,
+    Input,
+    InputGroup,
+    InputLeftElement,
+} from "@chakra-ui/react";
+import { useMemo, useState } from "react";
+import { FaFilter } from "react-icons/fa";
+import { useTable, useSortBy } from "react-table";
+
+const C2Table = (props) => {
+    const { columns, entries } = props;
+
+    const [data, setData] = useState(entries);
+
+    const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
+        useTable(
+            {
+                columns,
+                data,
+            },
+            useSortBy
+        );
+
+    const onFilter = (column, val) => {
+        // if e[column]
+        val = val.toLowerCase();
+
+        if (typeof data[0][column] !== "object") {
```

Comment:
When you type something that yields no results into the filter, deleting the entered value gives the error 
"Unhandled Runtime Error 
TypeError: Cannot read properties of undefined (reading 'course')"